using-packages.md

```diff
@@ -0,0 +1,262 @@
+   your app.
+
+## Managing package dependencies & versions
+
+### Package versions
+
+All packages have a version number, specified in their `pubspec.yaml` file. Pub
+displays the current version of a package next to it's name (for example, see
+the [url_launcher](https://pub.dartlang.org/packages/url_launcher) package), as
+well as a list of all prior versions ([url_launcher
+versions](https://pub.dartlang.org/packages/url_launcher#pub-pkg-tab-versions)).
+
+When a package is added to `pubspec.yaml` using the shorthand form `plugin1:`
+this is interpreted as `plugin1: any`, i.e. any version of the package may be
+used. To ensure your app does not break when a package is updated, we recommend
+specifying a version range using one of the following formats:
+
+* Range constraints: Specify a minimum and maximim version, e.g.:
+```dart
+dependencies:
+    url_launcher: '>=0.1.2 <0.2.0'
+```
+
+* Range constraint with [*caret syntax*](https://www.dartlang.org/tools/pub/dependencies#caret-syntax): 
+Similar to a regular range constraints
+```dart
+dependencies:
+    collection: '^0.1.2'
+```
+
+For additional details, see the [Pub versioning guide](https://www.dartlang.org/tools/pub/versioning).
+
+### Updating package dependencies
+
+When you run `flutter packages get` ('Packages Get' in IntelliJ) for the first
+time after adding a package, Flutter saves the concrete package version found in
+the `pubspec.lock`
+[lockfile](https://www.dartlang.org/tools/pub/glossary#lockfile). This ensures
+that you get the same version again if you, or another developer on your team,
+run `flutter packages get`.
+
+If you want to upgrade to a new version of the package, for example to use new
+features in that package, run `flutter packages upgrade` ('Upgrade dependencies'
+in IntelliJ). This will retrive the highest available version of the package,
+which is allowed by the version constraint you have specified in `pubspec.yaml`.
+
+### Dependencies on unpublished packages
+
+Packages can be used even when not published on Pub. For private plugins not
+intended for public publishing, or for packages not yet ready for publishing,
+additional dependency options are avaialble:
+
+* **Path** dependency: A Flutter app can depend on a plugin via a file system
+ `path:` dependency. The path can be either relative, or absolute. For example, to
+ depend on a plugin 'plugin1' located in a directory next to the app, use this
+ syntax:
+```dart
+dependencies:
+    plugin1:
+      path: ../plugin1/
+```
+
+* **Git** dependency: You can also depend on a package stored in a Git
+ repository. The package must be located in the root of the repo. Use this
+ syntax:
+```dart
+dependencies:
+    plugin1:
+      git:
+        url: git://github.com/flutter/plugin1.git
+```
+
+For more details, see the [Pub Dependencies article](https://www.dartlang.org/tools/pub/dependencies).
+
+## Examples
+
+### Example: Using the CSS Colors package {#css-example}
```

Comment:
You say "CSS Colors package" and then you start talking about URL Launcher. Is that intentional?